corehq/apps/custom_data_fields/static/custom_data_fields/js/custom_data_fields.js

```diff
@@ -266,10 +266,10 @@ hqDefine('custom_data_fields/js/custom_data_fields', [
                 JSON.stringify(option.value) === JSON.stringify(currentProfileRequiredForList || [])
             );
             initialRequiredFor = _.has(profileReqiredForMatch, 'value') ? profileReqiredForMatch.value :
-            (self.profileRequiredForOptions.find(option => option.isDefault)).value || [];
+            (self.profileRequiredForOptions.find(function (option) {return option && option.isDefault;}) || {}).value || [];
```

Comment:
I think we can do just `(self.profileRequiredForOptions.find(function (option) {return option.isDefault;}) || {}).value || [];` since we know  from`self.profileRequiredForOptions = options.profile_required_for_options || [];` that `profileRequiredForOptions` is either going to be an empty list or contain valid options (since it's something we already define)